code/modules/mob/living/carbon/human/life/handle_regular_hud_updates.dm

```diff
@@ -95,16 +95,28 @@
 			healths.icon_state = "health7" //DEAD healthmeter
 		return
 	else
-		change_sight(removing = SEE_TURFS|SEE_MOBS|SEE_OBJS)
+		// Moiving this "see invisble" thing here so that it can be overriden by xrays, vampires...
 
-		var/datum/organ/internal/eyes/E = src.internal_organs_by_name["eyes"]
-		if(E)
-			see_in_dark = E.see_in_dark //species.darksight
-		else
-			see_in_dark = species.darksight
+		// Vampire bandaid. I'm sorry.
+		var/datum/role/vampire/V = isvampire(src)
```

Comment:
This plus the block below = Absolutely heretical.

Reply:
Then find a way to do it that's not retarded.
The role process() is called before the life().

Reply:
Honestly this entire bit needs a rewrite, why the hell is sight change being called every life tick? Makes no sense.